Generals/Code/GameEngine/Source/GameClient/InGameUI.cpp

```diff
@@ -5424,42 +5424,37 @@ void InGameUI::selectNextIdleWorker( void )
 	if(getSelectCount() == 0 || getSelectCount() > 1)
 	{
 		selectThisObject = *m_idleWorkers[index].begin();
+		// If our idle worker is contained by anything, we need to select the container instead.
+		while (selectThisObject->getContainedBy())
+			selectThisObject = selectThisObject->getContainedBy();
 	}
 	else
 	{
 		Drawable *selectedDrawable = TheInGameUI->getFirstSelectedDrawable();	
-		
-		ObjectListIt it = m_idleWorkers[index].begin();
-		while(it != m_idleWorkers[index].end())
+		std::vector<Object*> uniqueIdleWorkers = getUniqueIdleWorkers(m_idleWorkers[index]);
+
+		ObjectPtrVector::iterator uit = uniqueIdleWorkers.begin();
```

Comment:
This is now the only iterator in this scope, so can name it back to `it` as before (makes a smaller diff and is more common name for iterator)